wdl-gauntlet/src/lib.rs

```diff
@@ -227,28 +228,35 @@ pub async fn gauntlet(args: Args) -> Result<()> {
                     term::emit(&mut buffer, &config, &file, &diagnostic.to_codespan())
                         .context("failed to write diagnostic")?;
 
+                    let byte_start = diagnostic
+                        .labels()
+                        .next()
+                        .map(|l| l.span().start())
+                        .unwrap_or_default();
+                    // The `+1` here is because line_index() is 0-based.
+                    let line_no = file.line_index((), byte_start).unwrap_or_default() + 1;
                     assert!(
-                        actual.insert(
+                        actual.insert((
                             std::str::from_utf8(buffer.as_slice())
                                 .context("diagnostic should be UTF-8")?
                                 .trim()
-                                .to_string()
-                        )
+                                .to_string(),
+                            line_no
+                        ))
                     );
                 }
             }
 
             // As the list of diagnostics has been sorted by document identifier, do
             // a binary search and collect the matching messages
             let diagnostics = config.inner().diagnostics();
-            let doc_id_str = document_identifier.to_string();
-            let expected: IndexSet<String> = diagnostics
-                .binary_search_by_key(&doc_id_str.as_str(), |d| d.document())
+            let expected: IndexSet<(String, usize)> = diagnostics
```

Comment:
Does this need to be `IndexSet<(String, usize)>`? Could it not just be `IndexSet<String>` containing only the messages?

That way below we don't need `expected_messages` either.